mango_library/negotiation/multiobjective_cohda/examples/simulation_util.py

```diff
@@ -435,43 +434,44 @@ def provide_schedules(solution_point=None, agent_id=None, candidate=None):
 
             agents.append(a)
             addrs.append((container.addr, a.aid))
-            schedules_per_agent[a.aid] = provide_schedules(agent_id=i+1)
+            schedules_per_agent[a.aid] = provide_schedules(agent_id=i + 1)
         # Controller agent will be a different agent, that is not part of the negotiation
         # Its tasks are creating a coalition and detecting the termination
-        controller_agent = RoleAgent(container)
+        controller_agent = container.register(RoleAgent())
         controller_agent.add_role(NegotiationTerminationDetectorRole())
-        controller_agent.add_role(
-            CoalitionInitiatorRole(
-                participants=addrs,
-                details="",
-                topic="",
-                topology_creator=topology_creator,
-            )
-        )
-        await asyncio.wait_for(wait_for_coalition_built(agents), timeout=5)
-        print("Done building a coalition.")
-
-        # fill the overlay dictionary
-        for a in agents:
-            assignment = next(
-                iter(
-                    a.roles[0]
-                    .context.get_or_create_model(CoalitionModel)
-                    ._assignments.values()
+        async with activate(container) as c:
```

Comment:
You could also use add_agent_composed_of and move schedule actions to on_ready() lifecycle method